crates/sha1/src/sha1.rs

```diff
@@ -0,0 +1,406 @@
+//! Circuits for the [SHA-1] hash function and its internal compression
+//! function.
+//!
+//! [SHA-1]: https://datatracker.ietf.org/doc/html/rfc3174
+
+use bellpepper::gadgets::{multieq::MultiEq, uint32::UInt32};
+use bellpepper_core::{boolean::Boolean, ConstraintSystem, SynthesisError};
+use ff::PrimeField;
+
+use crate::util::{and_uint32, or_uint32};
+
+const ROUND_CONSTANTS: [u32; 4] = [0x5a827999, 0x6ed9eba1, 0x8f1bbcdc, 0xca62c1d6];
+
+const IV: [u32; 5] = [0x67452301, 0xefcdab89, 0x98badcfe, 0x10325476, 0xc3d2e1f0];
+
+pub fn sha1_block_no_padding<Scalar, CS>(
+    mut cs: CS,
+    input: &[Boolean],
+) -> Result<Vec<Boolean>, SynthesisError>
+where
+    Scalar: PrimeField,
+    CS: ConstraintSystem<Scalar>,
+{
+    assert_eq!(input.len(), 512);
+
+    Ok(sha1_compression_function(&mut cs, input, &get_sha1_iv())?
+        .into_iter()
+        .flat_map(|e| e.into_bits_be())
+        .collect())
+}
+
+pub fn sha1<Scalar, CS>(mut cs: CS, input: &[Boolean]) -> Result<Vec<Boolean>, SynthesisError>
```

Comment:
The function could return an array instead of a `Vec` since the size is known at compile-time.

Reply:
I will try changing the output type here as well